src/extension/prompts/node/agent/test/agentPrompt.spec.tsx

```diff
@@ -233,4 +234,179 @@ suite('AgentPrompt', () => {
 				enableCacheBreakpoints: true,
 			})).toMatchSnapshot();
 	});
+
+	suite('TerminalAndTaskStatePromptElement', () => {
+		test('Copilot terminals and tasks', async () => {
+			const tasksService: any = {};
+			const terminalService: any = {};
+
+			tasksService.getTasks = () => [[null, [
+				{
+					label: 'npm: build',
+					isBackground: false,
+					type: 'npm',
+					command: 'build',
+					script: 'build',
+					problemMatcher: ['matcher1'],
+					group: { isDefault: true, kind: 'build' },
+					dependsOn: 'prebuild',
+				},
+				{
+					label: 'npm: watch',
+					isBackground: true,
+					type: 'npm',
+					command: 'watch',
+					script: 'watch',
+					problemMatcher: [],
+					group: { isDefault: false, kind: 'test' },
+				},
+			]]];
+			tasksService.isTaskActive = () => true;
+
+			terminalService.terminals = [
+				{ name: 'Terminal 1', id: '1' },
+				{ name: 'Terminal 2', id: '2' },
+			];
+			terminalService.getCopilotTerminals = async () => [
+				{ name: 'Terminal 1', id: '1' },
+				{ name: 'Terminal 2', id: '2' },
+			];
+			terminalService.getLastCommandForTerminal = (term: { id: string }) => {
+				if (term.id === '1') {
+					return { commandLine: 'npm run build', cwd: '/workspace', exitCode: 0 };
+				} else if (term.id === '2') {
+					return { commandLine: 'npm test', cwd: '/workspace', exitCode: 1 };
+				}
+				return undefined;
+			};
+
+			const prompt = new TerminalAndTaskStatePromptElement({}, tasksService, terminalService);
+			const rendered = await prompt.render();
+			const output = typeof rendered === 'string' ? rendered : JSON.stringify(rendered) ?? '';
+			assert(output.includes('Active Tasks:'));
```

Comment:
This is fine but the snapshots that I use in the other tests in this file are a really nice way to validate the whole output, that it's in the right order and formatting, and doesn't both include this string and `No active tasks found` and things like that, you might want to look at that if you aren't familiar with them, I've find them very useful.